Fix issue where wordpress config isn't read/written in correct directory, fixes #76

jakeparis opened the pull request:
when docroot isn't empty

Close #76

## The Issue

- #76 

<!-- Provide a brief description of the issue. -->

## How This PR Solves The Issue

Make the install/uninstall scripts aware of the ddev docroot if one is set

## Manual Testing Instructions

1. create a new ddev WordPress project, with a subdirectory docroot
2.  `ddev config --project-type=wordpress --docroot=html`
3. `ddev wp core download --path=html`
4. `ddev launch` and follow prompts 

(At this point, if you install the browsersync addon from it's main source, you'll see that the config files are incorrectly written to the top-level directory rather than to the docroot. You can't successfully remove the addon from this state either because it fails on removing things. You should restart these testing instructions on a fresh install)

5. install the addon `ddev add-on get https://github.com/jakeparis/ddev-browsersync/tarball/main`
6. `ddev restart`
7. The `wp-config-ddev-browsersync.php` file should be written into the docroot, next to `wp-config.php`
8. Extra config lines should be written into the `wp-config.php` file
   ```php
         /** Include for ddev-browsersync to modify WP_HOME and WP_SITEURL. */
      $ddev_browsersync_settings = dirname( __FILE__ ) . "/wp-config-ddev-browsersync.php";
      if ( is_readable( $ddev_browsersync_settings ) ) {
          require_once( $ddev_browsersync_settings );
      }
   ```

## Automated Testing Overview

Didn't change tests

## Release/Deployment Notes

No

rfay commented:
Thanks for the PR!

Please add more to your manual testing instructions, maybe provide a demo repo to work against? Currently the instructions just say to install the add-on, which certainly won't demonstrate the problem or the fix on its own.

jakeparis commented:
> Thanks for the PR!
> 
> Please add more to your manual testing instructions, maybe provide a demo repo to work against? Currently the instructions just say to install the add-on, which certainly won't demonstrate the problem or the fix on its own.

Of course, no problem. Updated now (and updated instructions for #79 as well)

tyler36 commented:
Thank you for the PR. 

I think the instruction are missing a step. 

```shell
ddev wp core download
ddev launch
```

Browser opens to a `403` error.  I'm not familiar enough with WP to move beyond that.

jakeparis commented:
> I think the instruction are missing a step.

Oh, you're right. The `ddev wp core download` needs a `--path=html` flag to put the files in the correct spot. I've updated the instructions and re-verified they work.

tyler36 commented:
Great. 

Was able to reproduce the error; "Could not process post-install action (3) 'Install browsersync settings for WordPress if applicable' "

This PR works well and adds `html/wp-config-ddev-browsersync.php`.

tyler36 commented:
Let's wait until #79 gets resolved so we know Browsersync is in a working state.

tyler36 reviewed (approved):
Looks good.

- Confirmed PR installs correctly following instructions.
- Confirmed `wp-config-ddev-browsersync.php` appears in `html` as expected.

## Test

1. Install following instructions
2. Run `ddev browersync` and view proxied site `...ddev.site:3000`
3. Edit a template (Eg. `html/wp-content/themes/twentytwentyfive/parts/header.html`) and confirm Browersync displays new content.
